examples/https_client/https_client.cc

```diff
@@ -0,0 +1,184 @@
+  SecureClient http_client{proactor};
+
+  http_client.set_connect_timeout_ms(2000);
+  auto list_res = proactor->Await([&] {
+    if (auto ec = http_client.Connect(host, service, ssl_ctx); !ec) {
+      h2::request<h2::string_body> req{h2::verb::get, resource, 11 /*http 1.1*/};
+      req.set(h2::field::host, std::string(host).c_str());
+      ResponseType res;
+      if (auto ec = http_client.Send(req, &res); !ec) {
+        return OpResult{RequestResults{res.result_int(), std::move(res)}};
+      } else {
+        return OpResult{nonstd::unexpected<std::string>("HTTPS GET Send ERROR: " + ec.message())};
+      }
+    } else {
+      return OpResult{nonstd::unexpected<std::string>("Connection error: " + ec.message())};
+    }
+    return OpResult{RequestResults{}};
+  });
+
+  return list_res;
+};
+
+using namespace std::chrono_literals;
+
+util::fibers_ext::Fiber ReadFromRemoteHost(ProactorPool* pool, SSL_CTX* ssl_context,
+                                           std::string_view task_name,
+                                           util::fibers_ext::Done* done_condition) {
+  std::string service = GetFlag(FLAGS_http_port);
+  std::string host = GetFlag(FLAGS_host);
+  std::string resource = GetFlag(FLAGS_resource);
+
+  LOG(INFO) << task_name << ": host '" << host << "', port " << service << ", resource '"
+            << resource << "'";
+  return pool->GetNextProactor()->LaunchFiber([=]() {
+    while (true) {
+      if (done_condition->WaitFor(1s)) {
+        VLOG(1) << task_name << ": finish running HTTPS task";
+        return;
+      }
+      auto pb = pool->GetNextProactor();
+
+      auto results = ConnectAndRead(pb, host, service, resource.c_str(), ssl_context);
+      if (results.has_value()) {
+        LOG(INFO) << task_name << ": successfully read";
+        LOG(INFO) << task_name << ": " << results.value();
+      } else {
+        LOG(WARNING) << task_name << ": Failed for '" << host << "':" << service << "/" << resource
+                     << ", error: " << results.error();
+      }
+    }
+  });
+}
+
+util::fibers_ext::Fiber IdleFiberRun(ProactorPool* pool, std::string_view task_name,
+                                     util::fibers_ext::Done* done_condition) {
+  return pool->GetNextProactor()->LaunchFiber([=]() {
+    while (true) {
+      VLOG(1) << task_name << ": just burning CPU";
+      if (done_condition->WaitFor(1s)) {
+        VLOG(1) << task_name << ": finish running with idle task";
+        return;
+      }
+    }
+  });
+}
+
+int main(int argc, char** argv) {
+  MainInitGuard guard(&argc, &argv);
+
+  std::string host = GetFlag(FLAGS_host);
+  if (host.empty()) {
+    LOG(ERROR) << "Host name is missing - please run with --host <host name>";
+    return -1;
+  }
+
+  SSL_library_init();
+  SSL_load_error_strings();
+  util::uring::UringPool pp;
+  int count = 0;
+  pp.Run();
+  SSL_CTX* ssl_context = SecureClient::CreateSslContext(SecureClient::LEVEL_2);
```

Comment:
who frees this context?

Reply:
You're right - I will add a "free" at the end